src/app/shared/localized-currency-pipe/localized-currency.pipe.ts

```diff
@@ -25,8 +25,10 @@ export class LocalizedCurrencyPipe implements PipeTransform, OnDestroy {
     });
   }
 
-  transform(value: any): any {
-    return `${value} ${LOCALIZED_CURRENCY[this.lang]}`;
+  transform(value: any, fixedDigits: boolean = false): any {
+    const formattedValue = fixedDigits ? (+value).toFixed(2) : Math.round(+value * 100) / 100;
+
+    return `${formattedValue} ${LOCALIZED_CURRENCY[this.lang]}`;
   }
 
   ngOnDestroy() {
```

Comment:
The destroy$.unsubscribe() call is unnecessary after emitting the complete event in ngOnDestroy:
```
  this.destroy$.complete();
```